pydantic/main.py

```diff
@@ -198,6 +199,13 @@ def validate_custom_root_type(fields: Dict[str, ModelField]) -> None:
         raise ValueError('__root__ cannot be mixed with other fields')
 
 
+def generate_hash_function(frozen: bool, super_class: Any) -> Optional[Callable[[Any], int]]:
+    def hash_function(self_: Any) -> int:
+        return hash(super_class) + hash(tuple(self_.__dict__.values()))
```

Comment:
IMO you could just do `hash(self_.__class__)` instead of using `super_class`

Reply:
Yes your are right ! https://github.com/samuelcolvin/pydantic/pull/1881/commits/12cff48de640035718d1308c04eb007fe5e1ef05 (I was overthinked this...)